Simplified/MyBooks/NYPLMyBooksDownloadCenter+AudiobooksDownload.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  NYPLMyBooksDownloadCenter+AudiobooksDownload.swift
+//  Simplified
+//
+//  Created by Ernest Fan on 2022-04-06.
+//  Copyright © 2022 NYPL. All rights reserved.
+//
+#if FEATURE_AUDIOBOOKS
+import Foundation
+import NYPLAudiobookToolkit
+
+extension NYPLMyBooksDownloadCenter {
+  /// Create an audiobook object and add to download queue.
+  /// Below are the steps:
+  ///   1. Retrieve book manifest from local storage
+  ///   2. Transform manifest into dictionary and add book identifier if needed
+  ///   3. Update vendor DRM key if needed
+  ///   4. Add audiobook to downloader queue
+  @objc(downloadAudiobookForBook:)
+  func downloadAudiobook(for book: NYPLBook?) {
+    guard let book = book,
+          let url = self.fileURL(forBookIndentifier: book.identifier) else {
+            Log.error(#file, "Missing book or url. Book: \(String(describing: book?.loggableShortString)). URL: \(String(describing:self.fileURL(forBookIndentifier: book?.identifier)))")
+            return
+    }
+    
+    let metadata = [
+      "book": book.loggableShortString(),
+      "fileURL": url.absoluteString
+    ]
+    
+    AudiobookManifestAdapter.transformManifestToDictionary(for: book,
+                                                              fileURL: url) { json, decryptor, error in
+      if error == .none,
+        let json = json {
+        AudioBookVendorsHelper.updateVendorKey(book: json) { [weak self] error in
+          guard let audiobook = AudiobookFactory.audiobook(json, decryptor: decryptor) else {
+            if let error = error {
+              NYPLErrorLogger.logError(error,
+                                       summary: "Audiobooks: DRM Error",
+                                       metadata: metadata)
+            } else {
+              Log.error(#file, "Audiobooks: unsupported - \(String(describing: book.loggableShortString))")
+            }
+            
+            // TODO: Handle Error, set state in book registry
```

Comment:
@ettore I am waiting for Risa on the behaviour of this failure. Will update once I heard back from her. Same for the one below.

Reply:
sounds good... btw generally speaking, feel free to file a ticket and do things you forgot on subsequent PRs if that's easier... after all we're merging to a feature branch, so you have a little more freedom :)